src/witness.rs

```diff
@@ -85,6 +85,11 @@ impl<F: Field> PartialWitness<F> {
         self.set_target(Target::Wire(wire), value);
     }
 
+    pub fn set_public_input(&mut self, pi: PublicInput, value: F) {
+        self.set_wire(pi.original_wire(), value);
+        self.set_target(pi.routable_target(), value);
```

Comment:
It should work without this since `PublicInputGate`'s generator copies the original wire to the routable wire, but also OK to keep if you prefer.

Reply:
Oh yeah, I think I added this to fix a bug, but you're right it's useless now 👍